src/github_app.py

```diff
@@ -0,0 +1,67 @@
+import os
+from datetime import datetime as dt, timedelta
+
+import jwt
+import requests
+
+
+def get_private_key():
+    private_key = None
+    if os.environ.get("GH_PRIVATE_KEY_PATH"):
+        # Open the file as f.
+        # The function readlines() reads the file.
+        with open(os.environ["GH_PRIVATE_KEY_PATH"], "rb") as f:
+            private_key = f.read()
+    else:
+        # XXX: On the next pass we need to support using Google Secrets
+        raise NotImplemented()
+
+    return private_key
+
+
+def get_jwt_token(private_key, app_id):
+    payload = {
+        # issued at time, 60 seconds in the past to allow for clock drift
+        "iat": dt.utcnow() - timedelta(minutes=1),
+        # JWT expiration time (10 minute maximum)
+        "exp": dt.utcnow() + timedelta(minutes=10),
+        # GitHub App's identifier
+        "iss": app_id,
+    }
+    return jwt.encode(payload, private_key, algorithm="RS256")
+
+
+def get_org_for_token(token):
+    headers = {
+        "Accept": "application/vnd.github.v3+json",
+        "Authorization": f"Bearer {token}",
+    }
+    req = requests.get("https://api.github.com/app", headers=headers)
+    req.raise_for_status()
+    response = req.json()
+
+    # The second value tracks the org this token is useful for
+    return response["owner"]["login"]
```

Comment:
this will get the owner of the *app*, I'm not sure if that's what's desired -- usually you want to get the installation by id or by org

this will coincidentally return the right thing if the app is created in the org and only used in the org however